source/_docs/deprecated_constructor_notices.md

```diff
@@ -0,0 +1,62 @@
+
+## Cause and Remediation
+
+This error is caused by a combination of two factors:
+
+- PHP notices generated during source code parsing
+- PHP opcache is in use
+
+To fix these errors, convert your class constructor to `__construct()`. See [deprecation notice for PHP-4-style constructors](http://php.net/manual/en/migration70.deprecated.php#migration70.deprecated.php4-constructors) for more information.
+
+## Finding PHP Files Containing Parse-Time Notices
+
+To make it easier to find and debug intermittent notices, **temporarily** disable the opcache in your settings.php file:
+```php
+if (!isset($_ENV['PANTHEON_ENVIRONMENT']) || ($_ENV['PANTHEON_ENVIRONMENT'] != 'live'))
+{
+    ini_set("opcache.enable", 0);
+}
+```
+**IMPORTANT:** Disabling opcache has a sever impact on performance, so care should be taken not to do this on a production system. Note that `display_errors` is set to `off` on Pantheon live environments, so there is no motivation to disable opcache in production anyway. When disabling opcache in dev and multi-dev environments, be sure to re-enable it again once you are done debugging.
+
+Note that you may also search for deprecated constructors using the PHP linter on the command line. First, create a local copy of your site as described in [Local Development](/docs/local-development); then, run:
+```nohighlight
+$ php -l sites/all/modules/views/includes/handlers.inc
+PHP Deprecated:  Methods with the same name as their class will not be constructors in a future version of PHP; views_many_to_one_helper has a deprecated constructor in sites/all/modules/views/includes/handlers.inc on line 753
+
+Deprecated: Methods with the same name as their class will not be constructors in a future version of PHP; views_many_to_one_helper has a deprecated constructor in sites/all/modules/views/includes/handlers.inc on line 753
+
+No syntax errors detected in sites/all/modules/views/includes/handlers.inc
+```
+Replace the path to the file you would like to check. To check every .php, .inc and .module file in a Drupal site, run:
+```nohighlight
+find -E . -iregex ".*\.(inc|php|module)$" -exec php -l {} \; | grep -v 'No syntax errors'
+```
+Adjust the regular expression as needed to scan other file extensions that may contain php code.
+
+## Explanation
+
+Deprecation notices for deprecated constructors are emitted if a class contains a method with the same name as the class name. This was the recommended way to declare class constructors in PHP 4, but it was not until PHP 7 that this form actually started producing a notice. At this time, the PHP-4-style constructor is the only deprecation notice that is reported during source code parsing. In theory, this structure should only be encountered in very old code; in practice, though, some projects may have continued using the deprecated form, so it may be encountered from time to time.
```

Comment:
Add an example of a class that contains a method with the same name as the class name?